Yona/Yona/Challenges/ChallengesFirstStepViewController.swift

```diff
@@ -11,65 +11,195 @@ import UIKit
 
 class ChallengesFirstStepViewController: UIViewController,UIScrollViewDelegate {
     
-    @IBOutlet var tegoedView: UIView!
-    @IBOutlet var tegoedBadgeLabel: UILabel!
+    enum SelectedCategoryHeader {
+        case BudgetGoal
+        case BudgetCategory
+        case TimezoneGoal
+        case TimezoneCategory
+        case NogoGoal
+        case NogoCategory
+    }
+    
+    @IBOutlet var gradientView: GradientView!
+    @IBOutlet var headerView: UIView!
+    @IBOutlet var headerLabel: UILabel!
+    
+    @IBOutlet var budgetView: UIView!
+    @IBOutlet var budgetBadgeLabel: UILabel!
     
-    @IBOutlet var tijdzoneView: UIView!
-    @IBOutlet var tijdzoneBadgeLabel: UILabel!
+    @IBOutlet var timezoneView: UIView!
+    @IBOutlet var timezoneBadgeLabel: UILabel!
     
     @IBOutlet var nogoView: UIView!
     @IBOutlet var nogoBadgeLabel: UILabel!
     
     @IBOutlet var scrollView: UIScrollView!
+    @IBOutlet var tableView: UITableView!
     
+    var selectedCategoryView: UIView!
+    var activityCategoriesArray = [Activities]()
+    var goalsArray = [Goal]()
     
+    var categoryHeader = SelectedCategoryHeader.BudgetGoal
+    
+    // MARK: - View
     override func viewDidLoad() {
         super.viewDidLoad()
+        
         setupUI()
     }
     
+    override func viewDidAppear(animated: Bool) {
+        gradientView.colors = [UIColor.yiSicklyGreenColor(), UIColor.yiSicklyGreenColor()]
+    }
+    
+    // MARK: - private functions
     private func setSelectedCategory(categoryView: UIView) {
+        selectedCategoryView = categoryView
         categoryView.addBottomBorderWithColor(UIColor.yiWhiteColor(), width: 4.0)
         categoryView.alpha = 1.0
+        if categoryView == budgetView {
+            categoryHeader = .BudgetGoal
+        } else if categoryView == timezoneView {
+            categoryHeader = .TimezoneGoal
+        } else if categoryView == nogoView {
+            categoryHeader = .NogoGoal
+        }
         
+        self.setHeaderTitleLabel()
     }
     
     private func setDeselectOtherCategory() {
-        tegoedView.addBottomBorderWithColor(UIColor.yiPeaColor(), width: 4.0)
-        tijdzoneView.addBottomBorderWithColor(UIColor.yiPeaColor(), width: 4.0)
+        budgetView.addBottomBorderWithColor(UIColor.yiPeaColor(), width: 4.0)
+        timezoneView.addBottomBorderWithColor(UIColor.yiPeaColor(), width: 4.0)
         nogoView.addBottomBorderWithColor(UIColor.yiPeaColor(), width: 4.0)
         
-        tegoedView.alpha = 0.5
-        tijdzoneView.alpha = 0.5
+        budgetView.alpha = 0.5
+        timezoneView.alpha = 0.5
         nogoView.alpha = 0.5
     }
     
+    private func callGoals() {
+        
+        APIServiceManager.sharedInstance.getUserGoals { (success, message, code, goals, error) in
+            if(success){
+                self.goalsArray  = goals!
```

Comment:
Use a closure here, do not force unwrap anything, where you have to use a ! you need to check the value first:
if let goals = goals {
   self.goalsArray = goals
}

This is a safe way of checking whether an optional parameter returned has a value...if you don't and no goals are returned the app will crash